semantic-python/test/Directive.hs

```diff
@@ -42,29 +59,62 @@ projects.
 -}
 data Directive = JQ ByteString -- | @# CHECK-JQ: expr@
                | Tree (Term Core Name) -- | @# CHECK-TREE: core@
+               | Result Text (Concrete (Term (Core.Ann Source.Span :+: Core)) Name) -- | @# CHECK-RESULT key: expected
                | Fails -- | @# CHECK-FAILS@ fails unless translation fails.
                  deriving (Eq, Show)
 
+-- | Extract all directives from a file.
+readDirectivesFromFile :: Path.Class.AbsRel ar => Path.File ar -> IO [Directive]
+readDirectivesFromFile
+  = runResourceT
+  . Stream.toList_
+  . Stream.mapM (either perish pure . parseDirective)
+  . Stream.takeWhile isComment
+  . Stream.mapped ByteStream.toStrict
+  . ByteStream.lines
+  . ByteStream.readFile @(ResourceT IO)
+  . Path.toString
+    where
+      perish s  = fail ("Directive parsing error: " <> s)
+      isComment = (== Just '#') . fmap fst . ByteString.uncons
+
+
 describe :: Directive -> String
-describe Fails = "<expect failure>"
-describe (Tree t) =  Core.Pretty.showCore t
-describe (JQ b) = ByteString.unpack b
+describe Fails        = "<expect failure>"
+describe (Tree t)     =  Core.Pretty.showCore t
+describe (JQ b)       = ByteString.unpack b
+describe (Result t e) = T.unpack t <> ": " <> show e
 
-fails :: Trifecta.Parser Directive
+fails :: CharParsing m => m Directive
```

Comment:
Switched to an abstract parser because we’re starting to use tokenization capabilities, and this makes modifying them easy.